src/nbl/asset/interchange/CGLTFLoader.h

```diff
@@ -0,0 +1,539 @@
+
+							bool validate()
+							{
+								for(auto& [attribute, accessor] : accessors)
+									if (!accessor.validate())
+										return false;
+
+								return true;
+							}
+						};
+
+						std::vector<SPrimitive> primitives;					//!< An array of primitives, each defining geometry to be rendered with a material.
+						std::optional<std::vector<uint32_t>> weights;		//!< Array of weights to be applied to the Morph Targets.
+						std::optional<std::string> name;					//!< The user-defined name of this object.
+
+						bool validate()
+						{
+							for (auto& primitive : primitives)
+								if (!primitive.validate())
+									return false;
+
+							return true;
+						}
+					};
+
+					SGLTFMesh glTFMesh;
+
+					bool validate(bool validateEntireTree = false)
+					{
+						if (mesh.has_value() && mesh.value() < 0)
+							return false;
+
+						if (camera.has_value() && camera.value() < 0)
+							return false;
+
+						if (skin.has_value() && skin.value() < 0)
+							return false;
+
+						if (validateEntireTree)
+							return glTFMesh.validate();
+						else
+							return true;
+					}
+				};
+
+				struct SGLTF
+				{
+					std::vector<SGLTFNode> nodes;
+
+					struct SGLTFBuffer
+					{
+						std::optional<std::string> uri;
+						std::optional<uint32_t> byteLength;
+						std::optional<std::string> name;
+
+						bool validate()
+						{
+							if (!uri.has_value())
+								return false;
+
+							if (!byteLength.has_value())
+								return false;
+							else
+								if (byteLength.value() < 1)
+									return false;
+
+							return true;
+						}
+					};
+
+					struct SGLTFBufferView
+					{
+						std::optional<uint32_t> buffer;
+						std::optional<uint32_t> byteOffset;
+						std::optional<uint32_t> byteLength;
+						std::optional<uint32_t> byteStride;
+						std::optional<uint32_t> target;
+						std::optional<std::string> name;
+
+						enum SGLTFTarget
+						{
+							SGLTFT_ARRAY_BUFFER = 34962,
+							SGLTFT_ELEMENT_ARRAY_BUFFER = 34963
+						};
+
+						bool validate()
+						{
+							if (!buffer.has_value())
+								return false;
+
+							if (!byteLength.has_value())
+								return false;
+
+							return true;
+						}
+					};
+
+					struct SGLTFImage
+					{
+						std::optional<std::string> uri;
+						std::optional<std::string> mimeType;
+						std::optional<uint32_t> bufferView;
+						std::optional<std::string> name;
+
+						struct SMIMEType
+						{
+							_NBL_STATIC_INLINE_CONSTEXPR std::string_view JPEG = "image/jpeg";
+							_NBL_STATIC_INLINE_CONSTEXPR std::string_view PNG = "image/png";
+						};
+
+						bool validate()
+						{
+							if (!uri.has_value() || (mimeType.has_value() && !bufferView.has_value()))
+								return false;
+
+							return true;
+						}
+					};
+
+					struct SGLTFSampler
+					{
+						std::optional<uint32_t> magFilter;
+						std::optional<uint32_t> minFilter;
+						std::optional<uint32_t> wrapS;
+						std::optional<uint32_t> wrapT;
```

Comment:
dont think they're ever optional, they just need sane defaults

Reply:
they aren't and have some defaults, but I can remove std::optional from there to let it be clear

Reply:
corrected in following commit